From cc28c6c2e6c579b244502c45ee72a966e123e8e8 Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@fb.com>
Date: Sun, 17 Apr 2016 16:39:25 -0600
Subject: [PATCH] blk-wb: trace fixes

- Log the right min latency
- Don't split read/write stats in two lines

Signed-off-by: Jens Axboe <axboe@fb.com>
---
 block/blk-wb.c               | 8 ++++----
 include/trace/events/block.h | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/block/blk-wb.c b/block/blk-wb.c
index 9e3afea..2e2da0a 100644
--- a/block/blk-wb.c
+++ b/block/blk-wb.c
@@ -192,11 +192,11 @@ static int __latency_exceeded(struct rq_wb *rwb, struct blk_rq_stat *stat)
 		maxlat = RWB_ROT_LAT;
 
 	if (stat[0].min > maxlat) {
-		trace_block_wb_lat(maxlat);
+		trace_block_wb_lat(stat[0].min);
 		trace_block_wb_stat(stat);
 		return LAT_EXCEEDED;
 	} else if ((thislat = rwb_sync_issue_lat(rwb)) > maxlat) {
-		trace_block_wb_lat(maxlat);
+		trace_block_wb_lat(thislat);
 		return LAT_EXCEEDED;
 	}
 
@@ -245,14 +245,14 @@ static void blk_wb_timer_fn(unsigned long data)
 	switch (status) {
 	case LAT_EXCEEDED:
 		step_down(rwb);
-		rwb_trace_step(rwb, "Down step");
+		rwb_trace_step(rwb, "down step");
 		break;
 	case LAT_OK:
 		if (rwb->scale_step) {
 			rwb->scale_step = 0;
 			rwb->perc = 100;
 			calc_wb_limits(rwb);
-			rwb_trace_step(rwb, "Reset");
+			rwb_trace_step(rwb, "reset");
 		}
 		break;
 	default:
diff --git a/include/trace/events/block.h b/include/trace/events/block.h
index 9e3d62c..1cca86d 100644
--- a/include/trace/events/block.h
+++ b/include/trace/events/block.h
@@ -702,8 +702,8 @@ TRACE_EVENT(block_wb_stat,
 		__entry->wnr_samples	= stat[1].nr_samples;
 	),
 
-	TP_printk("   read lat: mean=%llu, min=%llu, max=%llu, samples=%llu\n"
-		  "  write lat: mean=%llu, min=%llu, max=%llu, samples=%llu\n",
+	TP_printk("read lat: mean=%llu, min=%llu, max=%llu, samples=%llu,"
+		  "write lat: mean=%llu, min=%llu, max=%llu, samples=%llu\n",
 		  __entry->rmean, __entry->rmin, __entry->rmax,
 		  __entry->rnr_samples, __entry->wmean, __entry->wmin,
 		  __entry->wmax, __entry->wnr_samples)
-- 
2.8.1

